File: priv/combined_metrics/samples/function_design/boolean_function_has_question_mark/bad/predicates.py
class User:
    def admin(self):
        return self.role == "admin"

    def check_permission(self, perm):
        return perm in self.permissions

    def active(self):
        return not self.banned and self.confirmed_at is not None

    def verified(self):
        return self.email_verified and self.phone_verified

    def expired(self):
        return self.expires_at < datetime.now()

    def valid_subscription(self):
        return self.subscription and not self.subscription.expired

File: priv/combined_metrics/samples/function_design/boolean_function_has_question_mark/bad/predicates.js
class User {
  admin() {
    return this.role === "admin";
  }

  checkPermission(perm) {
    return this.permissions.includes(perm);
  }

  active() {
    return !this.banned && this.confirmedAt != null;
  }

  verified() {
    return this.emailVerified && this.phoneVerified;
  }

  expired() {
    return this.expiresAt < new Date();
  }

  getSubscriptionValid() {
    return this.subscription && !this.subscription.expired;
  }
}
